ktor-shared/ktor-shared-serialization-jackson/jvm/src/JacksonConverter.kt

```diff
@@ -44,9 +45,20 @@ public class JacksonConverter(private val objectmapper: ObjectMapper = jacksonOb
     }
 
     override suspend fun deserialize(charset: Charset, typeInfo: TypeInfo, content: ByteReadChannel): Any? {
-        return withContext(Dispatchers.IO) {
-            val reader = content.toInputStream().reader(charset)
-            objectmapper.readValue(reader, objectmapper.constructType(typeInfo.reifiedType))
+        try {
+            return withContext(Dispatchers.IO) {
+                val reader = content.toInputStream().reader(charset)
+                objectmapper.readValue(reader, objectmapper.constructType(typeInfo.reifiedType))
+            }
+        } catch (deserializeFailure: Exception) {
+            val badRequestException = BadRequestException("Illegal json parameter found", deserializeFailure)
+
+            when (deserializeFailure) {
+                is JsonParseException -> throw badRequestException
+                is IOException -> throw badRequestException
```

Comment:
Not sure about `IOException`. It is thrown on network errors as well. Why do you handle it as a bad request here?

Reply:
You are right, I removed it